be/src/olap/memory/column_reader.h

```diff
@@ -0,0 +1,95 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "olap/memory/column.h"
+
+namespace doris {
+namespace memory {
+
+// Holder for a ColumnBlock
+// Although ColumnBlock support reference counting, we avoid using it because
+// a reader already hold a proxy to all the ColumnBlocks, it's unnecessary
+// and inefficient to inc/dec refcount, so we use this holder instead.
+//
+// If the underlying column data doesn't need merge-on-read, we can use the
+// underlying base's ColumnBlock directly, and _own_cb equals false.
+//
+// If there are some deltas need to be merged, a new ColumnBlock will be
+// created, and _own_cb equals true.
+class ColumnBlockHolder {
```

Comment:
Is this thread-safe? Better declare it in the comment.

Reply:
fixed